lib/kafo/puppet_command.rb

```diff
@@ -44,5 +44,13 @@ def modules_path
           @configuration.kafo_modules_dir,
       ].flatten.join(':')
     end
+
+    def puppet_path
+      (::ENV['PATH'].split(File::PATH_SEPARATOR) + ['/opt/puppetlabs/bin']).find do |path|
```

Comment:
I know it's unlikely that another puppet would be installed during default values parsing but I think this should be cached so that the same puppet binary is called for both puppet runs, an example can be found at https://github.com/theforeman/kafo_parsers/blob/792d0096741d830966e9456b44f5d8891f6c2e16/lib/kafo_parsers/puppet_strings_module_parser.rb#L57

since `#find` is lazy, the return was not necessary here if you use `|| 'puppet'`

maybe something like this would work

```ruby
def puppet_path
  @puppet_path ||= begin
    bin_name = 'puppet'
    bin_path = (::ENV['PATH'].split(File::PATH_SEPARATOR) + ['/opt/puppetlabs/bin']).find do |path|
      File.executable?(File.join(path, bin_name))
    end
    File.join([bin_path, bin_name].compact)
  end
end
```